### PR TITLE
omnibus: python: fix extensions build

### PR DESCRIPTION
**Pull request by chouquette**
### What does this PR do?

Stop forwarding whatever compiler we use to the end user

### Motivation

If the compiler is only available on the build image, there's virtually no chance for the customers to have it available, which would cause failures when they build native python extensions

### Describe how to test/QA your changes

Ensure the following commands are successful:
```
docker run -it <image built by the pipeline> /bin/bash
apt-get update && apt-get install rrdtool librrd-dev build-essential -y
pip install rrdtool
```

### CODE REVIEW

**Review by alopezz — APPROVED** [review]
Some extra context for future viewers:

- We somewhat recently changed the Linux compiler toolchain to be able to target glibc versions regardless of the glibc version in the build environment's OS. That results in a change to the compiler name (from `gcc` to a more specific name related to the toolchain).
- setuptools relies on [`sysconfig.get_config_vars()`](https://docs.python.org/3/library/sysconfig.html#sysconfig.get_config_vars) to decide the default compiler to use when building extensions. This function extracts that information from the Makefile being modified by what was added in this PR.
- Said Makefile gets populated at Python build time by whatever compiler was used to build Python itself.
- With the cross compiling toolchain's compiler, the result was that installing Python libraries with native extensions in the Agent's embedded Python would attempt to use that compiler name, which will normally not exist in arbitrary hosts, as opposed to `gcc` which is far more common and the actual value that we had before the changes.
- Customers need to install such libraries to use some specific integrations where those libraries can't be shipped with the Agent (due to licensing reasons mostly).

File: releasenotes/notes/fix_python_extension_build-4891da3d4754769f.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Fix building of python extension containing native code.
```
Comment (urseberry) [inline releasenotes/notes/fix_python_extension_build-4891da3d4754769f.yaml:11]:
```suggestion
    Fix building of Python extension containing native code.
```

**Review by urseberry — COMMENTED** [review]
Capitalization nit.

**Review by hestonhoffman — APPROVED** [review]

**Comment by chouquette** [comment]
Added `qa-done` as this was manually QAed by @nubtron

**Comment by 0intro** [comment]
This change doesn't seem to work for me when building with the agent build image:

```
sed -i "s/${CC}/gcc/g" /opt/datadog-agent/embedded/lib/python3.12/_sysconfigdata__linux_*-linux-gnu.py
sed: -e expression #1, char 0: no previous regular expression
```

On my side, I had to do this change:

```
--- a/omnibus/config/software/python3.rb
+++ b/omnibus/config/software/python3.rb
@@ -57,8 +57,8 @@ if ohai["platform"] != "windows"
     if linux_target?
       command "sed -i \"s/^CC=[[:space:]]*${CC}/CC=gcc/\" #{install_dir}/embedded/lib/python#{major}.#{minor}/config-3.12-*-linux-gnu/Makefile", :env => env
       command "sed -i \"s/^CXX=[[:space:]]*${CXX}/CC=g++/\" #{install_dir}/embedded/lib/python#{major}.#{minor}/config-3.12-*-linux-gnu/Makefile", :env => env
-      command "sed -i \"s/${CC}/gcc/g\" #{install_dir}/embedded/lib/python#{major}.#{minor}/_sysconfigdata__linux_*-linux-gnu.py", :env => env
-      command "sed -i \"s/${CXX}/g++/g\" #{install_dir}/embedded/lib/python#{major}.#{minor}/_sysconfigdata__linux_*-linux-gnu.py", :env => env
+      command "sed -i \'s/${CC}/gcc/g\' #{install_dir}/embedded/lib/python#{major}.#{minor}/_sysconfigdata__linux_*-linux-gnu.py", :env => env
+      command "sed -i \'s/${CXX}/g++/g\' #{install_dir}/embedded/lib/python#{major}.#{minor}/_sysconfigdata__linux_*-linux-gnu.py", :env => env
     end
     delete "#{install_dir}/embedded/lib/python#{major}.#{minor}/test"
     block do
```

**Pull request merged**